### PR TITLE
Simplify overloads using optional params

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@JosephWoodward @jnm2 thoughts on this one. it is a breaking api change. so if we dont push it in 4, we will need to defer till 5

File: src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs [hunk 1]
```diff
@@ -492,16 +477,11 @@ public static class ShouldBeStringTestExtensions
             "stomMessage.", true)]
         public static void ShouldNotMatch(this string actual, string regexPattern, System.Func<string?>? customMessage) { }
         public static void ShouldNotMatch(this string actual, string regexPattern, string? customMessage = null) { }
-        public static void ShouldNotStartWith(this string? actual, string expected, Shouldly.Case caseSensitivity) { }
-        [System.Obsolete("Func based customMessage overloads have been removed. Pass in a string for the cu" +
-            "stomMessage.", true)]
-        public static void ShouldNotStartWith(this string? actual, string expected, System.Func<string?>? customMessage, Shouldly.Case caseSensitivity = 1) { }
```
Comment (jnm2) [inline src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs:498]: We should probably bring this back for consistency.

File: src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs [hunk 2]
```diff
@@ -583,12 +561,10 @@ public static void ShouldBeGreaterThanOrEqualTo<T>([System.Diagnostics.CodeAnaly
...
         public static void ShouldBeGreaterThanOrEqualTo<T>([System.Diagnostics.CodeAnalysis.AllowNull] this T actual, [System.Diagnostics.CodeAnalysis.AllowNull] T expected, System.Collections.Generic.IComparer<T> comparer, string? customMessage = null) { }
-        public static void ShouldBeInRange<T>(this T actual, T from, T to)
-            where T : System.IComparable<T> { }
         [System.Obsolete("Func based customMessage overloads have been removed. Pass in a string for the cu" +
             "stomMessage.", true)]
         public static void ShouldBeInRange<T>(this T actual, T from, T to, System.Func<string?>? customMessage) { }
-        public static void ShouldBeInRange<T>([System.Diagnostics.CodeAnalysis.DisallowNull] this T actual, [System.Diagnostics.CodeAnalysis.AllowNull] T from, [System.Diagnostics.CodeAnalysis.AllowNull] T to, string? customMessage)
+        public static void ShouldBeInRange<T>([System.Diagnostics.CodeAnalysis.DisallowNull] this T actual, [System.Diagnostics.CodeAnalysis.AllowNull] T from, [System.Diagnostics.CodeAnalysis.AllowNull] T to, string? customMessage = null)
```
Comment (jnm2) [inline src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs:567]: I wonder why were NRT attributes on one overload and not the other.

**Review by jnm2 — COMMENTED** [review]
I think this is a great time to make these improvements to API consistency.

File: src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs (hunk 2 above)
Comment (jnm2) [reply src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs:567]: Followed this to https://github.com/shouldly/shouldly/pull/687/files#r506340416.

File: src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs (hunk 1 above)
Comment (jnm2) [reply src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs:498]: https://github.com/shouldly/shouldly/pull/699

**Pull request merged**